crates/tests/mqtt_tests/src/test_mqtt_server.rs

```diff
@@ -9,7 +9,7 @@ use librumqttd::{Broker, Config, ConnectionSettings, ConsoleSettings, ServerSett
 use once_cell::sync::Lazy;
 use rumqttc::QoS;
 
-const MQTT_TEST_PORT: u16 = 55555;
+pub const MQTT_TEST_PORT: u16 = 55555;
```

Comment:
No, this internal constant must not be exposed because subject to change in favor a dynamic mechanism.

Use the port of the test mqtt broker.

```rust
let broker = mqtt_tests::test_mqtt_broker();

let client = create_test_mqtt_client(broker.port, mapper_config);
```

Reply:
> No, this internal constant must not be exposed because subject to change in favor a dynamic mechanism.
> 
> Use the port of the test mqtt broker.
> 
> ```rust
> let broker = mqtt_tests::test_mqtt_broker();
> 
> let client = create_test_mqtt_client(broker.port, mapper_config);
> ```

This solved the hanging tests. Without a broker, client cannot be created I guess.